### PR TITLE
Data validation

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn ready for review

File: go-app-clinic.js [hunk 1]
```diff
@@ -83,13 +121,34 @@ go.app = function() {
...
+                question = error;
+            }
+
             return new FreeText(name, {
-                question: $('Please input the mobile number of the ' +
-                            'pregnant woman to be registered:'),
+                question: question,
+
+                check: function(content) {
+                    if (!self.check_valid_number(content)) {
```
Comment (smn) [inline go-app-clinic.js:140]: Please cut a ticket for proper MSISDN validation capabilities in jsbox. No need to do that here though.

Comment (imsickofmaps) [inline go-app-clinic.js:140]: Done at https://github.com/praekelt/vumi-jssandbox-toolkit/issues/181

**Comment by smn** [comment]
For a shared lib of functions what I did on another project was:

screenshot_2014_04_24__10_33_am

File: src/personal.js [hunk 2]
```diff
@@ -115,11 +163,34 @@ go.app = function() {
             });
         });
 
-        self.states.add('states:sa_id', function(name) {
+        self.states.add('states:sa_id', function(name, opts) {
+            var error = $('Sorry, your ID number did not validate. ' +
+                          'Please reenter your SA ID number:');
```
Comment (smn) [inline src/personal.js:168]: re-enter? Not sure. You're english so you're probably right.

**Comment by smn** [comment]
:+1: in general, happy to see the shared function refactoring happen in a separate branch.

**Pull request merged**